src/Features/Core/Portable/EditAndContinue/Deltas.cs

```diff
@@ -4,18 +4,47 @@
 
 using System;
 using System.Collections.Immutable;
+using System.Xml;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.EditAndContinue
 {
     internal sealed class Deltas
     {
+        public sealed class Data
+        {
+            public Guid Mvid;
+            public ImmutableArray<byte> IL;
+            public ImmutableArray<byte> Metadata;
+            public ImmutableArray<byte> Pdb;
+            public ImmutableArray<(string SourceFilePath, ImmutableArray<(int OldLine, int NewLine)> Deltas)> LineEdits;
+            public ImmutableArray<int> UpdatedMethods;
+            public ImmutableArray<(ActiveMethodId Method, NonRemappableRegion Region)> NonRemappableRegions;
+            public ImmutableArray<(Guid ThreadId, Guid OldModuleId, int OldMethodToken, int OldMethodVersion, int OldILOffset, LinePositionSpan NewSpan)> ActiveStatementsInUpdatedMethods;
```

Comment:
might want to call this SerializableData (if the intent is to serialize it).  that would make it immediately obvious why it's all public/mutable fields.

Reply:
This will go away when we switch to message-pack.